tools/clang/test/HLSLFileCheck/hlsl/classes/ThisReference.hlsl

```diff
@@ -0,0 +1,42 @@
+// RUN: %dxc -T cs_6_6 -E main -ast-dump %s | FileCheck %s
+// RUN: %dxc -T cs_6_6 -E main -DERROR_CASE 2>&1 %s | FileCheck -check-prefix=FAIL %s
```

Comment:
This could also be accomplished with `FileCheck -input-file=stderr`, but perhaps that's something we want to get away from to better integrate with the LLVM style testing.